thamos/lib.py

```diff
@@ -97,7 +101,30 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
-def _wait_for_analysis(status_func: Callable[..., Any], analysis_id: str) -> None:
+def get_analysis_status(analysis_id: str, source: str) -> bool:
+    """Handle the the multiple response types available while asking for result of a anaylsis result."""
+    config = Configuration()
+    host = thoth_config.explicit_host
+    if not host:
+        thoth_config.load_config()
+        host = thoth_config.content.get("host") or config.host
+    source_url = _SOURCE.get(source)
```

Comment:
I think the `source` parameter is redundant. It can be obtained based on prefix from `analysis_id`.